extension/ts/main/ui/tooltip.ts

```diff
@@ -0,0 +1,36 @@
+import "../../../sass/tooltip.sass";
+
+import {UIColour} from "./colour";
+
+enum TooltipPosition {
+	TOP = "vbl-tool-tip-top",
+	BOTTOM = "vbl-tool-tip-bottom",
+	LEFT = "vbl-tool-tip-left",
+	RIGHT = "vbl-tool-tip-right",
+}
+
+interface TooltipOptions {
+	colour?: UIColour;
+	position?: TooltipPosition;
+	text: string;
+}
+
+const tooltipped = <T extends HTMLElement>(
+	element: T,
+	{text, colour = UIColour.GREY, position = TooltipPosition.TOP}: TooltipOptions
+): T => {
+	// This MiiIIIght not age well. Probably smarter to add an invisible anchor point inside element
```

Comment:
i h8 css -- if it breaks we can revisit?

Reply:
yeah i think we just would need another intermediate container